pinc/Project.inc

```diff
@@ -186,6 +186,7 @@ class ProjectPage
  * @property-read bool $PPVer_is_current_user
  * @property-read bool $forum_url
  * @property-read string $round_type
+ * @property-read string $round_document
```

Comment:
Thanks for elucidating this. I must admit I just followed what was there before, I should really look into the reasons for it. I used to be more familiar with a strongly typed language (C++) and imagined PHP had been hacked together for someone's Personal Home Page without any theoretical basis and it unfortunately became popular. Attempts were then made to add some rigour to it.

Reply:
> Thanks for elucidating this. I must admit I just followed what was there before, I should really look into the reasons for it. 

It's probably cleaner to avoid adding to this list... but PHPStan is clever enough to know the type of `$project->round_document` based on the annotation on Project and by inspecting  `Project::__get` so we get type coverage regardless of the path taken. All this to say: do what you prefer for now, we will clean it later if there is real reason :smile:

Reply:
> Do you know why we're defining those derived properties as a "magic" properties of an object rather than adding some public getter?

Because treating them like attributes is simpler than creating ~20 separate getters. I agree it's less ideal from a typing perspective.